### ISSUE TITLE
Display formatting: represent [datimeoffset] instances by their .ToString() value

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#18255]
### Summary of the new feature / enhancement

Related:
 * #3172
 * #13312
 * #13598

`[datetimeoffset]` offers many advantages of `[datetime]`, and is therefore generally preferable.

One obstacle to its wider adoption is its unwieldy _display formatting_, which currently  - due to the absence of formatting data - simply provides `Format-List` formatting for all properties of a `[datetimeoffset` instance, e.g.:

```powershell
# UNDESIRED
PS> [datetimeoffset]::Now

DateTime      : 10/11/2022 12:08:42 AM
UtcDateTime   : 10/11/2022 4:08:42 AM
LocalDateTime : 10/11/2022 12:08:42 AM
Date          : 10/11/2022 12:00:00 AM
Day           : 11
DayOfWeek     : Tuesday
DayOfYear     : 284
Hour          : 0
Millisecond   : 886
Microsecond   : 4
Nanosecond    : 0
Minute        : 8
Month         : 10
Offset        : -04:00:00
Second        : 42
Ticks         : 638010437228860040
UtcTicks      : 638010581228860040
TimeOfDay     : 00:08:42.8860040
Year          : 2022
```

A much more reasonable default representation would  be to simply use the `.ToString()` representation, which is a single-line representation that helpfully includes the UTC offset; e.g.:

```powershell
PS> [datetimeoffset]::Now.ToString()  # This should be the default representation; note the UTC-offset suffix.
10/11/2022 12:10:39 AM -04:00
```

Alternatively, make the representation more verbose to include the weekday and month _name_, as with `[datetime]`'s default output formatting, but the important part is that the UTC offset be included.

**Comment by SteveL-MSFT** [comment]
The current behavior is only if you literally display just a `[datetimeoffset]`, right?  When it's a member of an object, then `ToString()` does get called.  Is there another use case besides just looking at a direct `[datetimeoffset]` instance?

**Comment by mklement0** [comment]
Good point, @SteveL-MSFT, but I still think the stand-alone rendering should be single-line only, just like for `[datetime]`.

Consider the following example, whose current output is unhelpful:

```powershell
, [pscustomobject] @{ Date  = [datetimeoffset]::now } * 10  | Select-Object -ExpandProperty Date
```

**Comment by SteveL-MSFT** [comment]
I agree, in general, `[datetimeoffset]` should work the same as `[datetime]` in terms of formatting.  Probably just need the WG to decide on this.

**Comment by xtqqczze** [comment]
@iSazonov Please reopen.

**Comment by tats-u** [comment]
@SteveL-MSFT 

> Probably just need the WG to decide on this.

Could you share its result? People can't send a PR without it.

> .ToString() value

We should not. It's different from the display result of `DateTime`:

```
PS> [DateTime]::Parse("2026-01-02T03:04:05")

2026年1月2日 3:04:05


PS> [DateTimeOffset]::Parse("2026-01-02T03:04:05+09:00").ToString()
2026/01/02 3:04:05 +09:00

PS> [DateTime]::Parse("2026-01-02T03:04:05").ToString()
2026/01/02 3:04:05
```
